.golangci.yml

```diff
@@ -1,129 +0,0 @@
-# v1.51.2
```

Comment:
@imiric, to answer https://github.com/grafana/k6/pull/3033#pullrequestreview-1403575421 in a separate thread, I haven't tested how the symlink will work on Windows, but they should just work, NTFS supports them: https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/

Though, now that I think about this, `make ci-like-lint` won't work at the moment :disappointed: https://github.com/grafana/k6/blob/9fa50b2d1f259cdccff5cc7bc18a236d31c345ac/Makefile#L12-L17

> Could we maybe avoid the symlink by making the action download k6's file from master?

Hmm :thinking: If the `.golangci.yml` file is in the root of k6 repo and we download it on every CI run, then `make ci-like-lint` will continue working in this repo, though not in other repos that we want to switch to this approach :thinking: 

However, we won't be able to change the `.golangci.yml` file in a PR and observe how it affects the linter check results :thinking: They will always be downloaded from `master`.

Reply:
Actually, my bad, `make ci-like-lint` works in the k6 repo, because the whole folder is mounted in `/app`, including both the symlink and the actual `.golangci.yml`. However, it might not work in other repos that use these linting rules

Reply:
I'm familiar with Windows symlinks, but they're an NTFS feature. I'm also concerned with how Git handles links across platforms, and [it's not straightforward](https://stackoverflow.com/a/11664406).

My point was that by using symlinks we complicate things substantially, so we should try to avoid them. If someone wants to develop k6 on Android on a FAT32-formatted SD card, we shouldn't make this more difficult for them.

Apparently [Git will probe if symlinks are supported, and set the value of `core.symlinks` appropriately](https://git-scm.com/docs/git-config#Documentation/git-config.txt-coresymlinks), but do we want to depend on this functionality? Does it work only on fresh clones, or on branch pulls as well?

Docker was also historically weird with symlink support, so I'm not quite sure if `make ci-like-lint` would work on macOS or Windows with this change. I personally don't find that make target useful, so I wouldn't mind removing it, actually.

So I'd strongly suggest we avoid using symlinks, and find an alternative way to make this work.

> However, we won't be able to change the `.golangci.yml` file in a PR and observe how it affects the linter check results :thinking_face: They will always be downloaded from `master`.

I think this is a minor issue compared to using a symlink. But it could be worked around by downloading from `$GITHUB_REF` if we're running in this repository, or from `master` otherwise.

Reply:
> If someone wants to develop k6 on Android on a FAT32-formatted SD card, we shouldn't make this more difficult for them.

This is a bit of a stretch... :sweat_smile: In the very unlikely event that someone wants to do that, or develop it on a Windows XP machine, or something like that, I'd argue that they can rely on the linting we have in our CI :sweat_smile: Being able to run `golangci-lint` locally is not strictly required for anything.

> I think this is a minor issue compared to using a symlink. But it could be worked around by downloading from `$GITHUB_REF` if we're running in this repository, or from `master` otherwise.

It is not that simple, since the custom `composite` action is checked out differently when you reuse in another repo with `uses:`, but there actually might be a way to have this working the way you want. I'll play around with `{{ github.action_ref }}` and see how things work out.

Reply:
I am sharing my opinion. 

I'm okay with symlinks if they continue working for 80% of the cases. Yes, they are unpleasant, but it sorts the source-of-true issue for the k6 repository without additional actions from contributors/maintainers, which is good.

>If someone wants to develop k6 on Android on a FAT32-formatted SD card, we shouldn't make this more difficult for them.

But having a symlink doesn't block them from developing k6. They can still do that if they have a git and some text editor. And they still could use the `golangci-lint --config .github/actions/lint/.golangci.lint`, right? We could even stress it via makefile. Sure, it's overcomplicating the `inline` warnings for LSP, but other workarounds are possible there.

Regarding the `make ci-like-lint`, we could eliminate it if the team agrees and nobody uses them. I've constantly used it when we have "reversed" config file, and each update of the `goalngci-lint` causes many issues. But nowadays, it's almost not the case.

However, maybe having a customer `k6-golangci-linter` docker image could simplify a bit the possibility of using the rules for the external repositories.

Reply:
OK, after stepping on a few rakes (including the quite annoying to pin down https://github.com/actions/runner/issues/2473 / https://github.com/actions/runner/issues/2525 :man_facepalming:), I think I've resolved this problem :crossed_fingers: :pray: 

`.golangci.yml` can continue living in the k6 repo's root folder, while the reusable GitHub action downloads the correct version of the file on every run.

@imiric, please resolve this thread if you like the current version.

Reply:
Thanks for taking the time @na-- :bow: